lib/lte_link_control/lte_lc.c

```diff
@@ -222,7 +222,9 @@ static void at_handler_cereg(const char *response)
 	}
 
 	/* Cell update event */
-	if (memcmp(&cell, &prev_cell, sizeof(struct lte_lc_cell))) {
+	if ((cell.id != prev_cell.id) || (cell.tac != prev_cell.tac) ||
+	    (cell.mcc != prev_cell.mcc) || (cell.mnc != prev_cell.mnc) ||
+	    (cell.phys_cell_id != prev_cell.phys_cell_id)) {
```

Comment:
The LTE LC API says, that for the `LTE_LC_EVT_CELL_UPDATE` event, only `cell.tac` and `cell.id` members are populated. Also +CEREG notifications do not have MCC, MNC or physical cell ID fields. Would it be enough to check only TAC and cell ID here?

And just out of curiosity, would changing `struct lte_lc_cell cell = {0};` to a `memset()` call also work? I'm not saying that you should change the implementation even if it would. 😄

Reply:
Good point, I'll remove the additional checks. 

> And just out of curiosity, would changing struct `lte_lc_cell cell = {0};` to a `memset()` call also work? I'm not saying that you should change the implementation even if it would.

I'm not sure. I would actually expect `lte_lc_cell cell = {0};` to zero all out, but I'm not 100% sure the compiler will always zero the padding bytes, so this is mostly out of an abundance of caution. A `memset()` would always do it, that's true.